### PR TITLE
Update email widget validation regex

### LINKED ISSUE #3663: email widget validation does not correctly validate email addresses
**Describe the bug**

The current validation for the email type widget has a simple regex that does not allow for newer types of domains, such as .name.

**To Reproduce**

1. Create a schema form with an email widget
2. Submit the form with a valid email address such as `test@domain.name`
3. You will see the error `Input must be valid email (something@domain.com)`

**Expected behavior**

The email validation should accept all valid email addresses.

**Screenshot**

Screenshot 2022-09-21 at 09 53 43

**Additional context**
This [stackoverflow question](https://stackoverflow.com/questions/46155/how-can-i-validate-an-email-address-in-javascript) discusses the issue and provides a regex that will handle a wider array of valid email addresses.

The [current regex](https://github.com/plone/volto/blob/master/src/helpers/FormValidation/FormValidation.js#L43) is:

```
/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
```

I am proposing to replace this regex with the one provided in the above stackoverflow answer to:

```
  /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
```

### PR DESCRIPTION
**Pull request by instification**
Fixes #3663 

Regex was taken from https://html.spec.whatwg.org/multipage/input.html#e-mail-state-(type=email)

Also included an additional test to show the longer domain TLD being correctly validated.

### CODE REVIEW

File: src/helpers/FormValidation/FormValidation.js
Comment (stevepiercy) [inline src/helpers/FormValidation/FormValidation.js:43]: We should add a comment about why we chose this version of a regex, and provide a link to its source. Your commit message is good for this.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
Let's get one more reviewer from someone who is more familiar with Volto than I before merging.

**Review by sneridagh — APPROVED** [review]

**Pull request merged**
